hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/CommonConfigurationKeys.java

```diff
@@ -59,6 +59,16 @@
       "ipc.client.rpc-timeout.ms";
   /** Default value for IPC_CLIENT_RPC_TIMEOUT_KEY. */
   public static final int IPC_CLIENT_RPC_TIMEOUT_DEFAULT = 120000;
+  /** Enable the experimental use of netty instead of nio. */
+  public static final String IPC_SERVER_NETTY_ENABLE_KEY =
```

Comment:
IMO, if they are backward compatible, we can consider enabling it by default in Hadoop 3.4.0, and perhaps disable it by default in lower branches.

Reply:
I guess by backward compatible what is meant is,

The Server checks if the incoming request is SSL enabled, if yes, it continues with SSL otherwise it resorts to non-SSL based communication.

If you meant the above, yes the goal is to eventually support the above behaviour. I guess this what Daryn meant in his original comment on supporting a QoS like functionality.

But in the first version I plan to get SSL working and reject a communication from a client that is not SSL enabled.

I will definitely change the patch to be backward compatible in a later revision.